command/agent/event_endpoint.go

```diff
@@ -57,7 +61,13 @@ func (s *HTTPServer) EventFire(resp http.ResponseWriter, req *http.Request) (int
 	}
 
 	// Try to fire the event
-	if err := s.agent.UserEvent(dc, event); err != nil {
+	if err := s.agent.UserEvent(dc, token, event); err != nil {
+		if strings.Contains(err.Error(), permissionDenied) {
```

Comment:
I see we do this in a few places and this code will work fine as-is. Is the the usual way of looking for a particular error or would a type assertion on the error sub-type be a little cleaner?